Source/Scene/CreditDisplay.js

```diff
@@ -150,19 +97,26 @@ define([
 
     function removeCreditDomElement(credit) {
         var element = credit.element;
-        if (defined(element)) {
-            var container = element.parentNode;
-            if (!credit.hasImage() && credit.showOnScreen) {
-                var delimiter = element.previousSibling;
-                if (delimiter === null) {
-                    delimiter = element.nextSibling;
-                }
-                if (delimiter !== null) {
-                    container.removeChild(delimiter);
-                }
+
+        var container = credit.showOnScreen ? element.parentNode : element.parentNode.parentNode;
+
+        if (credit.hasImage() || !credit.showOnScreen) {
+            if (credit.showOnScreen) {
+                container.removeChild(element);
+            } else {
+                container.removeChild(element.parentNode);
             }
-            container.removeChild(element);
+            return;
+        }
+
+        var delimiter = element.previousSibling;
```

Comment:
This delimiter stuff should be in an `if (!credit.hasImage() && credit.showOnScreen)` block.  There aren't delimiters for the lightbox credits

Reply:
The above if statement already takes care of that and early returns.